crates/primitives/src/proofs.rs

```diff
@@ -104,20 +101,25 @@ pub fn calculate_receipt_root_optimism(
     ordered_trie_root_with_encoder(receipts, |r, buf| r.encode_inner(buf, false))
 }
 
+/// Calculates the receipt root for a header.
+pub fn calculate_receipt_root_ref(receipts: &[ReceiptWithBloomRef<'_>]) -> B256 {
```

Comment:
sigh, would be nice to have one method for both cases, unsure if it's possible though